File: tools/sap/src/main/java/top/bettercode/summer/tools/sap/connection/pojo/ISapReturn.java
package top.bettercode.summer.tools.sap.connection.pojo;

import com.fasterxml.jackson.annotation.JsonIgnore;

/**
 * @author Peter Wu
 */
public interface ISapReturn {

  @JsonIgnore
  default boolean isOk() {
    return isSuccess();
  }

  @JsonIgnore
  boolean isSuccess();

  @JsonIgnore
  String getMessage();
}

File: natives/sap/src/main/kotlin/top/bettercode/summer/tools/sap/connection/pojo/ISapReturn.kt
package top.bettercode.summer.tools.sap.connection.pojo

import com.fasterxml.jackson.annotation.JsonIgnore

/**
 * @author Peter Wu
 */
interface ISapReturn {
    @get:JsonIgnore
    val isOk: Boolean
        get() = isSuccess

    @get:JsonIgnore
    val isSuccess: Boolean

    @get:JsonIgnore
    val message: String?
}
